typescript/src/actions/evm/transfer/networkScopedTransfer.ts

```diff
@@ -0,0 +1,71 @@
+import { encodeFunctionData, erc20Abi } from "viem";
+
+import { getErc20Address } from "./utils.js";
+
+import type { Network, TransferOptions } from "./types.js";
+import type { EvmAccount } from "../../../accounts/evm/types.js";
+import type { Hex } from "../../../types/misc.js";
+import type { TransactionResult } from "../sendTransaction.js";
+import type { WalletClient } from "viem";
+
+/**
+ * Transfer an amount of a token from a network-scoped account to another account.
+ * This function is used for accounts that are scoped to a specific network and use
+ * a wallet client for transaction execution instead of the API client.
+ *
+ * @param walletClient - The wallet client to use for transaction execution.
+ * @param network - The network to transfer the token on.
+ * @param from - The account to send the transaction from.
+ * @param transferArgs - The transfer options.
+ * @returns The result of the transfer.
+ */
+export async function networkScopedTransfer<T extends EvmAccount>(
```

Comment:
I'm thinking a better name for this function could be `transferWithViem`. cc @sddioulde for thoughts here

Reply:
transferWithViem makes sense to me